core/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java

```diff
@@ -116,8 +130,23 @@ public Query termQuery(Object value, @Nullable QueryShardContext context) {
         @Override
         public Query termsQuery(List<?> values, @Nullable QueryShardContext context) {
             if (indexOptions() != IndexOptions.NONE) {
-                // 6.x index, _id is indexed
-                return super.termsQuery(values, context);
+                failIfNotIndexed();
+                BytesRef[] bytesRefs = new BytesRef[values.size()];
+                for (int i = 0; i < bytesRefs.length; i++) {
+                    BytesRef id;
+                    if (context.indexVersionCreated().onOrAfter(Version.V_6_0_0_alpha3)) {
```

Comment:
hmm maybe it's worth to move the if check out of the loop and assign it a boolean? I also wonder why the context is nullable? this looks like asking for trouble here?